src/ansys/speos/core/generic/visualization_methods.py

```diff
@@ -0,0 +1,285 @@
+
+        Parameters
+        ----------
+        y_vector: List[float]
+            The y-axis of the coordinate system.
+
+        Returns
+        -------
+        None
+        """
+        import pyvista as pv
+
+        if len(y_vector) != 3:
+            raise ValueError("y_axis must be a list with three elements.")
+        self._y_axis = pv.Arrow(
+            start=self._origin,
+            direction=normalize_vector(vector=y_vector),
+            scale=magnitude_vector(vector=y_vector),
+            tip_radius=0.05,
+            shaft_radius=0.01,
+        )
+
+    @property
+    def z_axis(self) -> "pv.Arrow":
+        """
+        Returns the z-axis of the coordinate system.
+
+        Returns
+        -------
+        pv.Arrow
+            pyvista.Arrow the z-axis of the coordinate system.
+
+        """
+        return self._z_axis
+
+    @z_axis.setter
+    def z_axis(self, z_vector: List[float]) -> None:
+        """Set the z-axis of the coordinate system.
+
+        Parameters
+        ----------
+        z_vector: List[float]
+            The z-axis of the coordinate system.
+
+        Returns
+        -------
+        None
+        """
+        import pyvista as pv
+
+        if len(z_vector) != 3:
+            raise ValueError("z_axis must be a list with three elements.")
+        self._z_axis = pv.Arrow(
+            start=self._origin,
+            direction=normalize_vector(vector=z_vector),
+            scale=magnitude_vector(vector=z_vector),
+            tip_radius=0.05,
+            shaft_radius=0.01,
+        )
+
+
+@graphics_required
+class _VisualData:
+    """Visualization data for the sensor.
+
+    By default, there is empty visualization data.
+
+    Notes
+    -----
+    **Do not instantiate this class yourself**
+    """
+
+    def __init__(self, coordinate_system=True):
+        import pyvista as pv
+
+        self._data = pv.PolyData()
+        self.coordinates = _VisualCoordinateSystem() if coordinate_system else None
+        self.updated = False
+
+    @property
+    def data(self) -> "pv.PolyData":
+        """
+        Returns the data of the sensor.
+
+        Returns
+        -------
+        pv.PolyData
+            The data of the surface visualization.
+
+        """
+        return self._data
+
+    def add_data_triangle(self, triangle_vertices: List[List[float]]) -> None:
+        """
+        Add surface data triangle to Visualization data.
+
+        Parameters
+        ----------
+        triangle_vertices: List[Vector]
+            The vertices of the triangle.
+
+        Returns
+        -------
+        None
+        """
+        import pyvista as pv
+
+        if len(triangle_vertices) != 3 or any(len(vertice) != 3 for vertice in triangle_vertices):
+            raise ValueError("provided triangle vertices must have 3 elements")
+        faces = [[3, 0, 1, 2]]
+        self._data = self._data.append_polydata(pv.PolyData(triangle_vertices, faces))
```

Comment:
```suggestion
        if len(triangle_vertices) != 3 or any(len(vertex) != 3 for vertex in triangle_vertices):
            raise ValueError("triangle_vertices is expected to be composed of 3 vertices with 3 elements each.")
        faces = [[3, 0, 1, 2]]
        self._data = self._data.append_polydata(pv.PolyData(triangle_vertices, faces))
```
Just wondering if 2D vertices shouldn't be accepted as they are accepted above ?

Reply:
2d vectices are not accepted here as mesh is always 3D. But, indeed, 2d vector calculation is allowed in the general_methods.py. Should I also limit the vector calculation to 3d as well?